packages/command/src/sfdxFlags.ts

```diff
@@ -45,14 +45,25 @@ function toValidatorFn(validator?: unknown): (val: string) => boolean {
   };
 }
 
+function merge<T>(
```

Comment:
Adding these overloads satisfies the different returns based on whether a boolean or option flag is provided. All flags should be the result of [`type IFlag<T> = IBooleanFlag<T> | IOptionFlag<T>`](https://github.com/oclif/parser/blob/master/src/flags.ts#L60) including sfdx's `flags.Any`
[`type Any<T> = Partial<OclifFlags.IFlag<T>> & SfdxProperties;`](https://github.com/forcedotcom/cli-packages/blob/b33f4490a11ab908344b84e8d40bf43342dd13b4/packages/command/src/sfdxFlags.ts#L86) which makes up the base definition.